src/Map.jl

```diff
@@ -0,0 +1,449 @@
+
+function show(io::IO, mime::MIME"text/plain", id::IdentityMap)
+    print(io, "idₘ: 𝑀 ⟶ 𝑀\n",
+              "𝑀 = $(repr(mime, domain(id)))")
+end
+
+@doc doc"""
+    Inclusion{D,CoD} <: AbstractMap{D,CoD}
+
+Construct the inclusion map $\iota$ from `D` to `CoD`, which identifies the
+input in the domain as belonging to the codomain. By default, this is just the
+identity map. However, it may be specialized to convert types of points where
+sensible.
+
+# Constructor
+
+    Inclusion(domain, codomain)
+"""
+struct Inclusion{D,CoD} <: AbstractMap{D,CoD}
+    domain::D
+    codomain::CoD
+end
+
+Inclusion(domain::D, ::D) where {D} = Identity(domain)
+
+(::Inclusion)(x) = identity(x)
+
+pinv(𝜄::Inclusion) = Inclusion(codomain(𝜄), domain(𝜄))
+
+function show(io::IO, mime::MIME"text/plain", 𝜄::Inclusion)
+    print(io, "𝜄: 𝑀 ↪ 𝑁\n",
+              "𝑀 = $(repr(mime, domain(𝜄)))\n",
+              "𝑁 = $(repr(mime, codomain(𝜄)))")
+end
+
+@doc doc"""
+    RiemannianExponential{MT,TMT,MMT} <: AbstractMap{TMT,MMT}
+
+Riemannian exponential map from a point on the tangent bundle $T M$ to
+to $M \times M$:
+
+```math
+\begin{aligned}
+\mathrm{Exp} &\colon T M \to M \times M \\
+             &\colon (x, v_x) \mapsto (x, y)\\
+\mathrm{where\ }& x, y \in M, v_x \in T_x M
+\end{aligned}
+```
+
+The abbreviated map name `Exp` is also available.
+
+# Constructor
+
+    RiemannianExponential(M::Manifold)
+"""
+struct RiemannianExponential{MT,TMT,MMT} <: AbstractMap{TMT,MMT}
+    manifold::MT
+    domain::TMT
+    codomain::MMT
+end
+
+const Exp = RiemannianExponential
+
+function RiemannianExponential(M::Manifold)
+    shape = representation_size(M, TVector)
+    TₓM = Euclidean(shape...)
+    TM = M × TₓM
+    M² = M × M
+    return RiemannianExponential(M, TM, M²)
+end
+
+(m::RiemannianExponential)(x, vₓ) = (x, exp(m.manifold, x, vₓ))
+(m::RiemannianExponential)(x::Tuple) = m(x...)
+(m::RiemannianExponential)(x) = ProductMPoint(x.parts[1], m(x.parts...))
+
+pinv(m::Exp) = Log(m.manifold)
+
+function show(io::IO, mime::MIME"text/plain", m::RiemannianExponential)
+    print(io, "Exp: 𝑇𝑀 ⟶ 𝑀 × 𝑀\n",
+              "𝑀 = $(repr(mime, m.manifold))")
+end
+
+@doc doc"""
+    RiemannianLogarithm{MT,MMT,TMT} <: AbstractMap{MMT,TMT}
+
+Riemannian logarithm map, which is the right inverse of the Riemannian
+exponential, defined as
+
+```math
+\begin{aligned}
+\mathrm{Log} &\colon M \times M \to T M\\
+             &\colon (x, y) \mapsto (x, v_x)\\
+\mathrm{Exp} \circ \mathrm{Log} &\colon (x, y) \mapsto (x, y)\\
+\mathrm{where\ }& x,y \in M, v_x \in T_x M
+\end{aligned}
+```
+
+The abbreviated map name `Log` is also available.
+
+# Constructor
+
+    RiemannianLogarithm(M::Manifold)
+"""
+struct RiemannianLogarithm{MT,MMT,TMT} <: AbstractMap{MMT,TMT}
+    manifold::MT
+    domain::MMT
+    codomain::TMT
+end
+
+const Log = RiemannianLogarithm
+
+function RiemannianLogarithm(M)
+    shape = representation_size(M, TVector)
+    TₓM = Euclidean(shape...)
```

Comment:
This will have to be rewritten once #31 is merged.

Reply:
Yep this is a placeholder (https://github.com/JuliaNLSolvers/Manifolds.jl/pull/28#issuecomment-515664204). :)